src/app/utils/api.ts

```diff
@@ -15,6 +16,49 @@ export function fetchJSON<T>(
   );
 }
 
-export function getMatches(): Promise<PaginatedMatches> {
-  return fetchJSON<PaginatedMatches>('/api/matches');
+export async function getMatches(): Promise<PaginatedMatchesClient> {
+  const matches = await fetchJSON<PaginatedMatchesClient>('/api/matches');
+  matches.results = matches.results.map((match) => ({
+    ...match,
+    createdAt: new Date(match.createdAt),
+  }));
+  return matches;
+}
+
+export async function postMatch(gameId: number): Promise<Match | null> {
+  const { username } = (await getCurrentUser()) || { username: 'unknown' };
+  const data = JSON.stringify({ gameId, username });
+
+  try {
+    return fetchJSON<Match>('/api/matches', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: data,
+    });
+  } catch (error) {
+    console.log(error);
+    return null;
+  }
+}
+
```

Comment:
Would it be useful, to extract the options object to use it in both functions? Propably something like this?
```suggestion

function createFetchOptions (data) {({
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: data,
    }
 )};

export async function postMatch(gameId: number): Promise<Match | null> {
  const { username } = (await getCurrentUser()) || { username: 'unknown' };
  const data = JSON.stringify({ gameId, username });

  try {
    return fetchJSON<Match>('/api/matches', createFetchOptions(data));
  } catch (error) {
    console.log(error);
    return null;
  }
}

```